controllers/argocd/configmap.go

```diff
@@ -651,48 +651,42 @@ func (r *ReconcileArgoCD) reconcileRedisConfiguration(cr *argoproj.ArgoCD, useTL
 // reconcileRedisHAConfigMap will ensure that the Redis HA Health ConfigMap is present for the given ArgoCD.
 func (r *ReconcileArgoCD) reconcileRedisHAHealthConfigMap(cr *argoproj.ArgoCD, useTLSForRedis bool) error {
 	cm := newConfigMapWithName(common.ArgoCDRedisHAHealthConfigMapName, cr)
-	if argoutil.IsObjectFound(r.Client, cr.Namespace, cm.Name, cm) {
-		if !cr.Spec.HA.Enabled {
-			// ConfigMap exists but HA enabled flag has been set to false, delete the ConfigMap
-			argoutil.LogResourceDeletion(log, cm, "redis ha is disabled")
-			return r.Client.Delete(context.TODO(), cm)
-		}
-		return nil // ConfigMap found with nothing changed, move along...
-	}
-
-	if !cr.Spec.HA.Enabled {
-		return nil // HA not enabled, do nothing.
-	}
-
 	cm.Data = map[string]string{
 		"redis_liveness.sh":    getRedisLivenessScript(useTLSForRedis),
 		"redis_readiness.sh":   getRedisReadinessScript(useTLSForRedis),
 		"sentinel_liveness.sh": getSentinelLivenessScript(useTLSForRedis),
 	}
+	if !cr.Spec.HA.Enabled {
+		// If HA is not enabled, delete the ConfigMap if it exists
+		existingCM := &corev1.ConfigMap{}
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, cm.Name, existingCM) {
+			return r.Client.Delete(context.TODO(), existingCM)
```

Comment:
```suggestion
                        argoutil.LogResourceDeletion(log, cm, "redis ha is disabled")
			return r.Client.Delete(context.TODO(), existingCM)
```
Can we log the deletion same as in the old code?